forest/cli/src/cli/snapshot_cmd.rs

```diff
@@ -0,0 +1,329 @@
+                        if let Some(date) = captures.name("date") {
+                            if let Ok(date) = time::Date::parse(date.as_str(), &Iso8601::DEFAULT) {
+                                if let Some(height) = captures.name("height") {
+                                    if let Ok(height) = height.as_str().parse::<i64>() {
+                                        if date > latest_date {
+                                            latest_date = date;
+                                            latest_height = height;
+                                            snapshot_to_keep = Some(path.clone());
+                                        } else if date == latest_date && height > latest_height {
+                                            latest_height = height;
+                                            snapshot_to_keep = Some(path.clone());
+                                        }
+
+                                        snapshots_with_valid_name.push(path);
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        if snapshots_with_valid_name.len() < 2 {
+            println!("No files to delete");
+            return;
+        }
+
+        let mut snapshots_to_delete = snapshots_with_valid_name;
+        let mut index_to_keep = 0;
+        println!("Files to delete:");
+        if let Some(snapshot_to_keep) = snapshot_to_keep {
+            for (i, path) in snapshots_to_delete.iter().enumerate() {
+                if &snapshot_to_keep != path {
+                    println!("{:?}", path.as_path());
+                } else {
+                    index_to_keep = i;
+                }
+            }
+        }
+        snapshots_to_delete.remove(index_to_keep);
+
+        if !yes && !prompt_confirm() {
+            println!("Aborted.");
+            return;
+        }
+
+        for snapshot_path in snapshots_to_delete {
+            delete_snapshot(&snapshot_path);
+        }
+    }
+}
+
+fn default_snapshot_dir(config: &Config) -> PathBuf {
+    config
+        .client
+        .data_dir
+        .join("snapshots")
+        .join(config.chain.name.clone())
+}
+
+fn delete_snapshot(snapshot_path: &PathBuf) {
+    let mut checksum_path = snapshot_path.clone();
+    checksum_path.set_extension("sha256sum");
+    for path in [snapshot_path, &checksum_path] {
+        if path.exists() {
+            if let Err(err) = fs::remove_file(path) {
+                println!("Failed to delete {:?}\n{err}", path);
+            } else {
+                println!("Deleted {:?}", path);
+            }
+        }
+    }
+}
+
+fn prompt_confirm() -> bool {
+    println!("Do you want to continue? [y/n]");
+    let mut line = String::new();
+    std::io::stdin().read_line(&mut line).unwrap();
+    let line = line.trim().to_lowercase();
+    line == "y" || line == "yes"
+}
```

Comment:
Can we use dialoguer API instead? It gives pretty colours and it's nicer not to re-implement the same thing.

https://github.com/mitsuhiko/dialoguer/blob/master/examples/confirm.rs

Reply:
It's built on top of console which does not work with `yes` tool, basically they are reading from terminal instead of stdin

Reply:
That's a bit disappointing, I expected such support from, I guess, the most popular crate for such operations. Oh well.